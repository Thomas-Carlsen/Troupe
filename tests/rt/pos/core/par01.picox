let
    fun loop x n =
      let fun body n =
            if n  > 0 then 
                let val dummy = printWithLabels x
                in
                    body (n-1)
                end
            else x
      in body n
      end


    val proc1 = fn () => loop 111 5
    
    val proc3 = fn () => loop 333 3

    val proc2 = fn () =>
                   let val dummy = spawn (proc3)
                   in loop 222 5
                   end
in let val pid2 = spawn (proc2)
    in let val x = printWithLabels pid2
        in proc1 ()
        end
    end
end
