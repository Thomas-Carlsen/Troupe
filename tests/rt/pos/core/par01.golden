2019-05-07T18:14:08.168Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
1e4f2230-2ca0-46a0-9a06-e2a9e24e04c9@{}%{}
111@{}%{}
111@{}%{}
111@{}%{}
111@{}%{}
111@{}%{}
>>> Main thread finished with value: 111@{}%{}
222@{}%{}
222@{}%{}
222@{}%{}
222@{}%{}
222@{}%{}
333@{}%{}
333@{}%{}
333@{}%{}
